src/frontend/js/components/AddressesManagement/useAddressesManagement.tsx

```diff
@@ -0,0 +1,135 @@
+import { messages } from 'components/AddressesManagement/index';
+import { AddressesMutateOptions, useAddresses } from 'hooks/useAddresses';
+import { useState } from 'react';
+import { useIntl } from 'react-intl';
+import { MutateOptions } from 'react-query';
+import * as Joanie from 'types/Joanie';
+import { Address } from 'types/Joanie';
+import { Maybe } from 'types/utils';
+import { HttpError } from 'utils/errors/HttpError';
+import { confirm } from 'utils/indirection/window';
+
+/**
+ * High-level helper to handle various `Address` CRUD with loading and error built-in.
+ */
+export function useAddressesManagement() {
+  const intl = useIntl();
+  const addresses = useAddresses();
+  const [error, setError] = useState<Maybe<string>>();
+  const [load, setLoad] = useState(false);
```

Comment:
As this is a boolean value, I prefer to prefix this state with a verb (`is`, `has`, ...)
Here I think `isLoading` makes sense